src/olympia/users/tests/test_commands.py

```diff
@@ -473,3 +478,128 @@ def test_migrate_twice(self):
+                domain=domain, reason=f'Disposable email domain of {provider}'
+            ).exists()
+
+        assert DisposableEmailDomainRestriction.objects.count() == len(expected)
+
+        os.remove(tmp_path)
+
+    def test_file_with_missing_columns(self):
+        """Test that rows with missing columns are ignored or handled gracefully."""
+        csv_content = (
+            'Domain,Provider\n'
+            'mailfast.pro,incognitomail.co\n'
+            ',mail.tm\n'  # Missing domain
+            'mailpro.live,incognitomail.co\n'
+        )
+
+        with tempfile.NamedTemporaryFile(mode='w+', delete=False, suffix='.csv') as tmp:
+            tmp.write(csv_content)
+            tmp.flush()
+            tmp_path = tmp.name
+
+        assert DisposableEmailDomainRestriction.objects.count() == 0
+
+        call_command('bulk_add_disposable_domains', tmp_path)
+
+        expected = [
+            ('mailfast.pro', 'incognitomail.co'),
+            ('mailpro.live', 'incognitomail.co'),
+        ]
+        for domain, provider in expected:
+            assert DisposableEmailDomainRestriction.objects.filter(
+                domain=domain, reason=f'Disposable email domain of {provider}'
+            ).exists()
+
+        assert not DisposableEmailDomainRestriction.objects.filter(domain='').exists()
+        assert DisposableEmailDomainRestriction.objects.count() == len(expected)
+
+        os.remove(tmp_path)
+
+    def test_file_with_header_only(self):
+        """Test that a file with only a header row does not trigger any additions."""
+        csv_content = 'Domain,Provider\n'
+
+        with tempfile.NamedTemporaryFile(mode='w+', delete=False, suffix='.csv') as tmp:
+            tmp.write(csv_content)
+            tmp.flush()
+            tmp_path = tmp.name
+
+        call_command('bulk_add_disposable_domains', tmp_path)
+
+        assert DisposableEmailDomainRestriction.objects.count() == 0
+
+        os.remove(tmp_path)
+
+    @patch('olympia.users.management.commands.bulk_add_disposable_domains.logger')
+    @patch(
+        'olympia.users.management.commands.bulk_add_disposable_domains.bulk_add_disposable_email_domains'
+    )
+    def test_bulk_add_result_is_printed(self, mock_bulk_add, mock_logger):
+        """result of bulk_add_disposable_email_domains task logged."""
+        fake_result = 'Task completed successfully'
+        mock_bulk_add.apply.return_value = fake_result
+
+        csv_content = (
+            'Domain,Provider\n'
+            'mailfast.pro,incognitomail.co\n'
+            'mailpro.live,incognitomail.co\n'
+        )
+
+        with tempfile.NamedTemporaryFile(mode='w+', delete=False, suffix='.csv') as tmp:
+            tmp.write(csv_content)
+            tmp.flush()
+            tmp_path = tmp.name
+
+        try:
+            call_command('bulk_add_disposable_domains', tmp_path)
+            mock_logger.info.assert_called_with(fake_result)
+        finally:
+            os.remove(tmp_path)
```

Comment:
Tbh I'm not sure what this is actually testing with the mocks - seems like it would be more accurate to just mock the logger and call the command with correct data to check it returns that value.  Or add that logging mock to one of the other tests.

Reply:
> seems like it would be more accurate to just mock the logger and call the command with correct data to check it returns that value

AFAICT that is literally what the test does.. we mock the data and the logger and assert the log gets passed `fake_result`

Reply:
No, it's mocking the task return too, so it's only testing that a mocked value is used on the following line.  You could set `fake_result` to be "BLAH BLAH"; or completely rewrite `bulk_add_disposable_email_domains` so something different is returned and it'd still pass.